crates/ruff_python_semantic/src/analyze/typing.rs

```diff
@@ -315,10 +315,12 @@ pub fn is_type_checking_block(stmt: &ast::StmtIf, semantic: &SemanticModel) -> b
     }
 
     // Ex) `if typing.TYPE_CHECKING:`
-    if semantic
-        .resolve_call_path(test)
-        .is_some_and(|call_path| matches!(call_path.as_slice(), ["typing", "TYPE_CHECKING"]))
-    {
+    if semantic.resolve_call_path(test).is_some_and(|call_path| {
+        matches!(
+            call_path.as_slice(),
+            ["typing", "TYPE_CHECKING"] | ["typing_extensions" | "TYPE_CHECKING"]
+        )
+    }) {
         return true;
```

Comment:
We may want to instead use `semantic.match_typing_call_path` which checks `typing_extensions`, but also `_typeshed` and any user-configured modules... Though I don't know if type checkers respect `TYPING_CHECKING` re-exports? I assume so.

Reply:
That's a good idea I'll look into that too